src/platform/intel/cavs/lib/clk.c

```diff
@@ -166,14 +166,9 @@ static void platform_clock_low_power_mode(int clock, bool enable)
 	int freq_idx = *cache_to_uncache(&active_freq_idx);
 
 	if (enable && current_freq_idx != waiti_freq_idx)
-		/* LPRO requests are fast, but requests for other ROs
-		 * can take a lot of time. That's why it's better to
-		 * not release active clock just for waiti,
-		 * so they can be switched without delay on wake up.
-		 */
-		select_cpu_clock(waiti_freq_idx, false);
```

Comment:
Can we add another comment here explaining the new logic.

Reply:
no logic changes in final version

Reply:
ping

Reply:
This comment was only about delay which isnt important for us, the remaining logic is pretty simple just as 'if' in code, if clock > LPRO then change to LPRO